### PR TITLE
add webhook validation on CUE template outputs name

### LINKED ISSUE #1093: check outputs:resource should be unique in application webhook
The trait can use `context.outputs.<xx>`  to do simple data passing.

This rely on the trait resource name is unique, or the former outputs resource data will be covered by the latter one.

We should add [a webhook validate rule](https://github.com/oam-dev/kubevela/tree/master/pkg/webhook/core.oam.dev/v1alpha2/application) to check for it.

https://github.com/oam-dev/kubevela/pull/1087/files#diff-05b89d808deaa1f5a7e06ea0b6de559283ab1e2ea28e006299af38a200fa9bcaR354

_Originally posted by @zzxwill in https://github.com/oam-dev/kubevela/pull/1092#r580846972_

### PR DESCRIPTION
**Pull request by captainroy-hy**
to fix #1093 

wait for merging #1447 
Signed-off-by: roy wang <seiwy2010@gmail.com>

### CODE REVIEW

File: pkg/appfile/validate.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+func newValidationProcessContext(wl *Workload, appName, revisionName, ns string) (process.Context, error) {
+	// add hook funcs to validate CUE base
+	baseHooks := []process.BaseHook{}
+	// add hook funcs to validate CUE auxiliaries
+	auxiliaryHooks := []process.AuxiliaryHook{validateAuxiliaryNameUnique()}
+
+	if wl.FullTemplate.Reference.Kind == "" {
+		// only validate output has GVK when workload referenced GVK is empty
```
Comment (wonderflow) [inline pkg/appfile/validate.go:61]:
```suggestion
		// validate output has GVK when workload referenced GVK is empty
```

only?

File: pkg/appfile/validate.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	if err := wl.EvalContext(pCtx); err != nil {
+		return nil, errors.Wrapf(err, "evaluate base template app=%s in namespace=%s", appName, ns)
+	}
+	return pCtx, nil
+}
+
+// validateOutputGVK validates whether the output entity in a
+// ComponentDefinition has a GVK after evaluation.
+func validateOutputGVK() process.BaseHook {
```
Comment (wonderflow) [inline pkg/appfile/validate.go:78]: I think this validation is necessary, the `wl.FullTemplate.Reference.Kind` will never be empty as it should be validate and mutate in webhook, this is a missing issue \cc @yangsoon

Comment (captainroy-hy) [reply pkg/appfile/validate.go:78]: sure, then I will make this validation compulsory.

**Review by wonderflow — APPROVED** [review]
Great job, LGTM after comments fixed

**Pull request merged**